pgxpool/pool.go

```diff
@@ -408,6 +417,9 @@ func (p *Pool) backgroundHealthCheck() {
 }
 
 func (p *Pool) checkHealth() {
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
```

Comment:
I opted for a `time.Ticker` on this one for simplicity, but this does have the side effect of changing this behavior from a "sleep 500ms between attempts" to a "make an attempt every 500ms" type behavior. If the health checks take awhile then there could be minimal sleeping time.

I can refactor to use a timer, it would just be messier so I wanted to propose this option first and see what you think.